### PR TITLE
Refactor some code using opts.compReloc 

### PR DESCRIPTION
**Pull request by briansull**
Refactor code for constant handles that reference opts.compReloc to use ImmedValNeedsReloc()

### CODE REVIEW

**Comment by briansull** [comment]
@dotnet-bot Windows_NT x86 full_opt legacy_backend CoreCLR Perf Tests Correctness

**Comment by briansull** [comment]
@dotnet/jit-contrib PTAL
This is a zero code diff change

File: src/jit/codegenlegacy.cpp [hunk 1]
```diff
@@ -11603,8 +11603,10 @@ void CodeGen::genCodeForTreeSmpOpAsg(GenTreePtr tree)
 
         case GT_CNS_INT:
 
+            GenTreeIntConCommon* con;
+            con = op2->AsIntConCommon();
```
Comment (sandreenko) [inline src/jit/codegenlegacy.cpp]: `GenTreeIntConCommon* con = op2->AsIntConCommon(); `

File: src/jit/gentree.cpp [hunk 2]
```diff
@@ -3045,7 +3045,8 @@ unsigned Compiler::gtSetEvalOrder(GenTree* tree)
     {
         switch (oper)
         {
-            bool iconNeedsReloc;
+            GenTreeIntConCommon* con;
```
Comment (sandreenko) [inline src/jit/gentree.cpp]: Is it better to define them in their own scopes?

Comment (briansull) [reply src/jit/gentree.cpp]: I will make this change in the next chaneet

**Pull request merged**